src/Security/UserChecker.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace AppBundle\Security;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Exception\AccountNotValidatedException;
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+use Symfony\Component\Security\Core\Exception\DisabledException;
+use Symfony\Component\Security\Core\User\UserCheckerInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class UserChecker implements UserCheckerInterface
+{
+    private $router;
+
+    public function __construct(Router $router)
+    {
+        $this->router = $router;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function checkPreAuth(UserInterface $user)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function checkPostAuth(UserInterface $user)
+    {
+        /** @var Adherent $user */
+        if (!$user instanceof Adherent) {
+            throw new \UnexpectedValueException('You have to pass an Adherent instance.');
+        }
+
+        if ($user->isEnabled()) {
+            return;
```

Comment:
To make things clearer, I would prefer to invert the test and throw the exception here.